tests/test_security.py

```diff
@@ -0,0 +1,105 @@
+"""
+In this module we test the basic ability to sign payloads and verify the signature.
+"""
+
+import base64
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
+from cryptography.exceptions import InvalidSignature
+
+import pytest
+
+from sqooler.security import (
+    JWSHeader,
+    JWSDict,
+    JWK,
+    payload_to_base64url,
+    sign_payload,
+    jwk_from_config_str,
+    create_jwk_pair,
+    public_from_private_jwk,
+)
+
+private_key = Ed25519PrivateKey.generate()
+public_key = private_key.public_key()
+
+
+def test_sign_payload() -> None:
+    """
+    Test the ability to sign a payload
+    """
+    payload = {"test": "test"}
+
+    header = JWSHeader(kid="test")
+    header_base64 = header.to_base64url()
+    payload_base64 = payload_to_base64url(payload)
+    full_message = header_base64 + b"." + payload_base64
+
+    signature = private_key.sign(full_message)
+    signature_base64 = base64.urlsafe_b64encode(signature)
+    jws_obj = JWSDict(header=header, payload=payload, signature=signature_base64)
+
+    # and now we can verify the signature
+    public_key.verify(signature, full_message)
+
+    # now also test that it fails with the wrong message
+    wrong_message = b"wrong message"
```

Comment:
maybe to test a wrong payload {"test": "test1"} with the correct header and signature. This is the case that someone tries to change a number in your result and the verification process could detect that.

Reply:
Added in 7e0454c